src/converter/one/convert/JfrConverter.java

```diff
@@ -21,31 +21,52 @@
 public abstract class JfrConverter extends Classifier {
     protected final JfrReader jfr;
     protected final Arguments args;
+    private final IEventAggregator leakAggregator;
     protected Dictionary<String> methodNames;
 
     public JfrConverter(JfrReader jfr, Arguments args) {
+        boolean leakDetection = args.nativemem && args.leak;
+
         this.jfr = jfr;
         this.args = args;
+        this.leakAggregator = leakDetection ? new MallocLeakAggregator(args.threads, args.total,
+                args.lock ? 1e9 / jfr.ticksPerSec : 1.0) : null;
     }
 
     public void convert() throws IOException {
         jfr.stopAtNewChunk = true;
         while (jfr.hasMoreChunks()) {
             // Reset method dictionary, since new chunk may have different IDs
             methodNames = new Dictionary<>();
-            convertChunk();
+            convertChunk(collectEvents());
+        }
+
+        if (leakAggregator != null) {
+            leakAggregator.finish();
+
+            if (args.grain > 0) {
+                leakAggregator.coarsen(args.grain);
+            }
+
+            convertChunk(leakAggregator);
```

Comment:
An illustration to the above comment: we have a special case for leakAggregator, but trying to deal with it as with some generic aggregator.

Can we design it in a way that we either have a common sequence for any aggregator or don't pretend that leak processor is an aggregator at all and just make leak processor explicitly a different thing?

For example, to consolidate EventAggregator (currently created for each new chunk) and LeakAggegator (a single instance for the whole process) we may probably replace creation of a new EventAggregator with clearing/resetting the same instance of EventAggregator. (Just as an idea; I'm open for other options too).

Reply:
Removing the leak and special-cased invocations.